test/lib/pack.js

```diff
@@ -158,6 +159,45 @@ t.test('should log pack contents', (t) => {
   })
 })
 
+t.test('should log output as valid json', (t) => {
+  const testDir = t.testdir({
+    'package.json': JSON.stringify({
+      name: 'my-cool-pkg',
+      version: '1.0.0',
+    }, null, 2),
+  })
+
+  const Pack = t.mock('../../lib/pack.js', {
+    libnpmpack: () => libpack(testDir),
```

Comment:
If this is just calling the real thing it doesn't need to be mocked, right?

Reply:
I have mixed feelings about that.

On the one hand, I would like to write to the mock for this function. On the other hand, to do this I need to create a `Buffer` containing the tarball content.

It looks like I can dive deeper and mock the [function](https://github.com/npm/cli/blob/049166bbc96b6e223a436dd849f3bf3daac37164/lib/pack.js#L69) of getting content from tarball.

Also I looked at the other implementations in the tests and found the following options:

* Use the `tar` package ([here](https://github.com/npm/cli/blob/049166bbc96b6e223a436dd849f3bf3daac37164/test/lib/publish.js#L285-L329)).
* Use the real `libnpmpack` package ([here](https://github.com/npm/cli/blob/049166bbc96b6e223a436dd849f3bf3daac37164/test/lib/utils/tar.js#L34-L39)).

Reply:
I replaced the real call with a mock of the tarball content reading function.

https://github.com/npm/cli/compare/48a349c66c607b50b1c209b32ef482d935d07cb4..d18c29e0a8571c48097bcf15265546a7d79585e9

Reply:
Yeah I just double checked and that test file is already mocking libnpmpack, which is not your problem to solve here.  

Can we test for all of the attributes returned in the assertion, such as size and mode of each file as well as id, name, version, size etc of the main package too?  The test really needs to tell the whole story here.

Reply:
took care of it, used info from a small real published package `@ruyadorno/redact` in order to have mocked info as close to real data as possible.